AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -81,6 +75,36 @@ - (void)channel:(id<MSChannelProtocol>)__unused channel prepareLog:(id<MSLog>)lo
   }
 }
 
+- (void)channel:(id<MSChannelProtocol>)channel
+     didPrepareLog:(id<MSLog>)log
+    withInternalId:(NSString *)__unused internalId {
+  if (![self shouldSendLogToOneCollector:log] || ![channel conformsToProtocol:@protocol(MSChannelUnitProtocol)]) {
+    return;
+  }
+  id<MSChannelUnitProtocol> channelUnit = (id<MSChannelUnitProtocol>)channel;
+  NSString *groupId = channelUnit.configuration.groupId;
+  id<MSChannelUnitProtocol> oneCollectorChannelUnit = [self.oneCollectorChannels objectForKey:groupId];
+  if (!oneCollectorChannelUnit) {
+    return;
+  }
+  id<MSLogConversion> logConversion = (id<MSLogConversion>)log;
+  NSArray<MSCommonSchemaLog *> *commonSchemaLogs = [logConversion toCommonSchemaLogs];
+  for (MSCommonSchemaLog *commonSchemaLog in commonSchemaLogs) {
+    [oneCollectorChannelUnit enqueueItem:commonSchemaLog];
+  }
+}
+
+- (BOOL)channelUnit:(id<MSChannelUnitProtocol>)__unused channelUnit shouldFilterLog:(id<MSLog>)log {
+
+  // Validate Custom Schema logs, filter out invalid logs.
+  if ([log isKindOfClass:[MSCommonSchemaLog class]]) {
+    return ![self validateLog:(MSCommonSchemaLog *)log];
+  }
+
+  // It's an App Center log. Filter out if it countains token(s) since it's already re-enqeued as CS log(s).
+  return [[log transmissionTargetTokens] count] > 0;
```

Comment:
Why do you check `transmissionTargetTokens` for App Center logs?

Reply:
This was not introduced in this story, code just moved :). But the reason is we want to stop processing AC logs with tokens after they got converted to CS logs.